telemetry/telemetrybuffer.go

```diff
@@ -299,17 +303,36 @@ func WaitForTelemetrySocket(maxAttempt int, waitTimeInMillisecs time.Duration) {
 }
 
 // StartTelemetryService - Kills if any telemetry service runs and start new telemetry service
-func StartTelemetryService(path string, args []string) error {
-	platform.KillProcessByName(TelemetryServiceProcessName)
+func (tb *TelemetryBuffer) StartTelemetryService(path string, args []string) error {
+	err := platform.NewExecClient(tb.logger).KillProcessByName(TelemetryServiceProcessName)
```

Comment:
Shouldn't this be added as exec client to the telemetry buffer ?

Reply:
KillProcessByName is added to execClient interface